lib/metrics.py

```diff
@@ -348,3 +346,77 @@ def _get_db_conn(self):
                 psycopg2.extensions.ISOLATION_LEVEL_AUTOCOMMIT
             )
         return self.db
+
+
+class PaidAppsMetricsEmitterThread(BaseMetricsEmitterThread):
+    @property
+    def _select_stats_to_emit(self):
+        selected_stats = []
+        if buildpackutil.i_am_primary_instance():
+            selected_stats = [
+                self._inject_database_stats,
+                self._inject_storage_stats,
+                self._inject_health,
+            ]
+        selected_stats.append(self._inject_m2ee_stats)
+        return selected_stats
+
+    def _gather_metrics(self):
+        stats = {}
+        try:
+            for inject_method in self._select_stats_to_emit:
+                stats = inject_method(stats)
+        except psycopg2.OperationalError as up:
+            logger.exception("METRICS: error while gathering metrics")
```

Comment:
What do you think of extending this error message with a bit more info relating the issue to the database?
I know that line numbers show up in the exception and thus in the logs so we can pinpoint where it came from, but it might be easier to distinguish this from the catchall clause in L377 if we use a more informative error message.

Reply:
Bear in mind that these all buildpack error messages show up in customer logs, and they result in tickets, because customers check logs, and open support tickets when there are errors.

Often we cannot do much about those errors.

We should be extremely cautious when modifying or adding extra buildpack logging, as it ends up in customer logs.

(The ideal situation is that we have our own private logs for internal buildpack logging, but that won't happen particularly soon)

Reply:
According to docs logger.exception already includes Exception info, so I don't think modifying log message is necessary. https://docs.python.org/3/library/logging.html#logging.exception

Please also notice that while I moved this logic from run to separate method don't want to change anything about the logic. 

As I have mentioned in some other discussion, for me personally it is weird that we decided to not include all other available metrics in case of exception while gathering a single category.

We already have broadened up the scope of this PR, outside of the original one, I don't think proper logging for Paid Apps falls under scope of this PR

What do you think?